lib/calculator/parser.rb

```diff
@@ -63,7 +63,61 @@ def shift_tokens
 
     # Entry point for the parser, parses an <em>expression</em>
     def parse_expression
+
       # TODO implement me
+      def parse_expression
+
+        ExpressionNode.new(parse_term, parse_expression_prime)
+      end
+
+      def parse_expression_prime
+            case @tokens.first
+              when AddOpToken
+                ExpressionPrimeNode.new(parse_term, parse_expression_prime, shift_tokens)
+              when SubtractOpToken
+                ExpressionPrimeNode.new(parse_term, parse_expression_prime, shift_tokens)
+
+              else
+                ExpressionPrimeNode.new
+
+            end
+        def parse_term
+            TermNode.new(parse_term_prime, parse_factor)
```

Comment:
Here again the rule is term -> factor term', so I think you need to reverse the order of the arguments.